src/Microsoft.AspNetCore.Authorization.Middleware/AuthorizationMiddleware.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authentication;
+using Microsoft.AspNetCore.Authorization.Policy;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    public class AuthorizationMiddleware
+    {
+        private readonly RequestDelegate _next;
+        private readonly IAuthorizationPolicyProvider _policyProvider;
+
+        public AuthorizationMiddleware(RequestDelegate next, IAuthorizationPolicyProvider policyProvider)
+        {
+            if (next == null)
+            {
+                throw new ArgumentNullException(nameof(next));
+            }
+
+            if (policyProvider == null)
+            {
+                throw new ArgumentNullException(nameof(policyProvider));
+            }
+
+            _next = next;
+            _policyProvider = policyProvider;
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+
+            var endpoint = context.Features.Get<IEndpointFeature>()?.Endpoint;
+            if (endpoint == null)
```

Comment:
how about a default policy?

Reply:
What do you mean by default? I get the impression from the filter code that the default comes into play when there is an attribute with no policy or roles, i.e. `[Authorize]`

If there is no endpoint then there is no `AuthorizeAttribute`, and so the default policy doesn't come into play. Correct: yay/nay?

Reply:
That's true sofar, but being middleware allows for a new scenario. For example lots of people write middleware today that denies anonymous requests, or applies policy like conditions. See the StaticFile example.

Above I proposed `UseAuthorization("default")` to opt into a specific policy regardless of the endpoint.

Reply:
So if there's no endpoint and no authorize attribute, the authZ middleware is a no-op?

Reply:
Right now, yes